src/main/java/com/example/book/controller/BookController.java

```diff
@@ -48,4 +50,9 @@ public BookDto updateBook(@PathVariable Long id, @RequestBody BookDto updateRequ
     public void deleteBook(@PathVariable Long id) {
         bookService.deleteBook(id);
     }
+
+    @GetMapping("/search")
+    public List<BookDto> searchBooks(@Valid BookSearchParametersDto searchParametersDto) {
```

Comment:
Remove `@Valid` here, you don't have any requirements to validate search dto
```suggestion
    public List<BookDto> searchBooks(BookSearchParametersDto searchParametersDto) {
```